_posts/en/newsletters/2020-07-15-newsletter.md

```diff
@@ -0,0 +1,131 @@
+  `SIGHASH_ANYPREVOUT` and `SIGHASH_ANYPREVOUTANYSCRIPT`.  Both
+  proposals describe optional signature hash (sighash) flags that do not
+  commit to the particular UTXOs (inputs/previous outputs) being spent
+  in a transaction, making it possible to create a signature for a
+  transaction without knowing what UTXO it will spend.  That feature can
+  be used by the proposed [eltoo][topic eltoo] settlement layer to allow
+  creating a series of transactions where any later transaction can
+  spend the value from any earlier transaction, allowing an offchain
+  contract to be settled in its latest state even if earlier states get
+  confirmed onchain.
+
+    The main difference between the noinput and anyprevout proposals is
+    that noinput would require its own new version of segwit but
+    anyprevout uses one of the upgrade features from the proposed [BIP342][]
+    specification of [tapscript][topic tapscript].  Additional
+    differences between the proposals are described in the [revised
+    text][anyprevout revisions] itself.
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+FIXME:none so far; harding to update or delete this section Tuesday
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1 repo],
+[Hardware Wallet Interface (HWI)][hwi], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #19219][] clarifies the distinction between manual peer banning
+  and automated peer discouragement, and it improves the latter's resource usage
+  by creating a non-persisted rolling Bloom filter of misbehaving peers to
+  prevent limited connection slots from being abused by them. Such peers are now
+  logged as *discouraged* rather than *banned* to reflect the reality that since
+  [Bitcoin Core #14929][] they are not actually banned; incoming connections
+  from them are still allowed, though they are preferred for eviction if the
+  inbound slots are full, and nodes don't make outgoing connections to them or
+  gossip their address. In contrast, incoming connections are never accepted
+  from manually banned peers, and their addresses and subnets are persisted to
+  banlist.dat on shutdown and reloaded on startup. Banning can be used to
+  prevent connections with spy nodes or other griefers.
+
+    It's important to note that neither banning nor discouragement protect
+    against DoS attacks, as it is trivial for an attacker to reconnect from
+    different IP addresses, and attempting to automatically disconnect or ban
+    any class of peer carries the risk of splitting the network.
```

Comment:
I think this write-up is really good. It's longer than we normally give for a merge summary, and if we shortened it, I'd recommend we cut between "In contrast ..." and "...splitting the network" (since no changes are actually made to the manual ban functionality).  An alternative could be to link to the code comment at https://github.com/bitcoin/bitcoin/blob/f4de89edfa8be4501534fec0c662c650a4ce7ef2/src/banman.h#L29-L55 for more details.

Reply:
That code comment is excellent (kudos to its author!) and I referenced it considerably; I wasn't sure what the Optech policy is WRT to links to commits.

Reply:
Links to commits and code and anything that's inherently factual are always great; feel free to pile them on.  Links to opinionated material (especially tweets, personal blog posts, etc) are something I'd lightly discourage but find acceptable if it's especially relevant.

Reply:
Thanks, SGTM